### ISSUE TITLE
Breadboard wiring 

### ISSUE THREAD

**Issue by Omegaleg** [samyk/combobreaker#14]
Hey Samyk can you post the breadboard schematic for the simpler combo breaker you showed in your video?

**Comment by mlemasters2002** [comment]
there have been many to ask for this in the past.  I know there has been some work on developing the PCB, that might help to understand.  We are about to start this project as well, just got the parts, but now comes the more difficult part.

**Comment by Omegaleg** [comment]
Good luck on building, can you send me a picture of the wiring once you are done. That would really help me out.

**Comment by mlemasters2002** [comment]
I will share all I can, I have been looking at the wiring schematic and will likely start working on it in the next few weeks.  There was once a branch on this where folks where building a PCB, but that seems to be cleaned up now.  If you start and have success let everyone know.

**Comment by AidanMcCarthy** [comment]
@mlemasters2002 @Omegaleg The branch with the PCB that I redesigned has been merged into the master branch under pull #13 .

Make sure to use the files under [https://github.com/samyk/combobreaker/tree/master/PCB/combobreaker_v2_oki_78sr_eagle](url)

You may well run into issues with the code once you've got the hardware setup. [I managed to get it working at a very basic level by making some changes](https://github.com/samyk/combobreaker/pull/13/commits/6d1320a91a40cb2a7e5487d147e57cca7abb74ca?file-filters%5B%5D=.ino#diff-825eaa5acc4bfe10efb9cae2b8edf170) but it seemed to only be bruteforcing and not using the algorithm. The stepper also didn't seem to have quite enough power to turn the dial if it was at all sticky or stiff.

**Comment by mlemasters2002** [comment]
Thank you Aidan, have you heard of anyone getting this working yet?  As a brute force tool, have affective has it been?

**Comment by AidanMcCarthy** [comment]
The best I've seen is [this one built by NYC CNC](https://www.youtube.com/watch?v=tteG2hQaDeg). I'm not aware of anyone that's successfully used this one to crack a lock.

**Comment by Omegaleg** [comment]
@AidanMcCarthy so I followed the wiring diagram I found on Imgur https://imgur.com/a/Ci0Kp and got the hardware set up. I tried using your basic code but it kept saying saying: "stray @ in program". I tried deleting the @ but it still does not work, Im still a beginner and only have a basic coding knowledge. Can you help me with the code, Im thinking of making this for my science project.

**Comment by mlemasters2002** [comment]
We Will likely start the build this weekend, can you send a photo of your breadboard as well, if we notice anything we will come back and post as well.

**Comment by Omegaleg** [comment]
@mlemasters2002 Good luck building.

**Comment by mlemasters2002** [comment]
This wiring diagram looks a bit different from the diagram in the official project, did you have any luck getting it to work?

**Comment by Omegaleg** [comment]
I finished the wiring, however the basic code doesn't work so Im still trying to figure out why.

**Comment by mlemasters2002** [comment]
We have it mostly wired now, will be working on it again hopefully tomorrow night.  Did you have any luck so far in getting the stepper and server running?

**Comment by Omegaleg** [comment]
Nope still working on code.

**Comment by mlemasters2002** [comment]
Tonight we had some limited success, we believe all the wiring is correct as in the stepper is turning both right and left and the servo is lifting on the shackle.  However the dial receiver is too lose and we will have to do a slight redesign.  Also, I think we lowered the stepper power a bit too low.  The shackle are also needs to be extended, before we can really test.  I have no idea yet if the optical encoder is working yet.  I am looking to putting on the display to see if any errors occur.

**Comment by Omegaleg** [comment]
So has anyone got it to work yet?

**Comment by mlemasters2002** [comment]
Have to remodel some parts to really let it run.  Everything seems to be working but have not yet gotten it to unlock as the reviewer is slipping on the lock.  How about you, how far have you gotten?

**Comment by Omegaleg** [comment]
I have wired everything onto the breadboard, but the simple code @AidanMcCarthy gave me still does not work.

**Comment by mlemasters2002** [comment]
What code is it, is it different from that in the project.  Do you have control of the servo and the stepper?

**Comment by mlemasters2002** [comment]
If you can send me the IMO, I will see if I can get it to work.  Does it compile?

**Comment by mlemasters2002** [comment]
Does anyone have the brute force code that Samy mentions.  I don’t see a flag to set to brute force in the code.  I have the same experience with the stepper, the alignment  is not the best in the printed part.  If we get it working properly we will re-design the fram and share it out with everyone.  Aiden did you get your PCB working, can it be purchased?

**Comment by AidanMcCarthy** [comment]
@mlemasters2002 DM me on Twitter at @Aidan_McCarthy

**Comment by Omegaleg** [comment]
This is the code https://github.com/samyk/combobreaker/pull/13/commits/6d1320a91a40cb2a7e5487d147e57cca7abb74ca?file-filters%5B%5D=.ino#diff-825eaa5acc4bfe10efb9cae2b8edf170

**Comment by Omegaleg** [comment]
So anyone got the code to work yet?

**Comment by mlemasters2002** [comment]
I didn’t get much done this weekend, will have to start up again this week.  I am hoping to have a new test by mid week.

**Comment by AidanMcCarthy** [comment]
While chatting with @mlemasters2002 last week I ran some tests as well and got this working on a basic level.

https://github.com/AidanMcCarthy/combobreaker/commit/20867e955b5b0feb889a8362496fe3feb1a4cf06

**Comment by mlemasters2002** [comment]
I noticed a new thing during the brut force, I need to record it and send it out.

**Comment by mlemasters2002** [comment]
I am not sure if my last e-mail attached, I just sent Aidan the video via twitter.  Here is what I did, I am about to finish the Proto board as well, only a few more wires to go.  

I could not upload to the project, here is the video.  I have modified my make shift hook arm.  Also, I have eighth step enable, I modified the code to fully put it back in,  I am not sure if the encoder is working, I wonder if there is a way to test if it is actually doing something. Notice at the end of the video what it is doing, does this seem right?  the receiver is slipping at the moment so it is all messed during that part of the cycle.

**Comment by mlemasters2002** [comment]
Here are the edits I am working on, this is not fully tested yet, but I will test tonight.

https://github.com/mlemasters2002/combobreaker/blob/master/Combobreaker_Revised_10-01-2019.ino

**Comment by mlemasters2002** [comment]
Should get some testing in this weekend on this, has anyone had any luck.  Aiden the new design is looking good. We will share out the new dial receiver this weekend.

**Comment by mlemasters2002** [comment]
So I am a GitHub idiot, but I started a new branch because I have not had time to really play around yet.  I uploaded the dial receiver as a f3d file so you can play with it.  Aidan can you share the new chassis files, would like to see if we can add a version for the servo.  
https://github.com/mlemasters2002/combobreaker/blob/master/MLCCC%20V3.1%20v2.f3d

**Comment by Omegaleg** [comment]
Im still having trouble with @AidanMcCarthy's code, here is the error message I am getting. And yes I did name the program danny devito. 

danny_devito:1:1: error: stray '@' in program
 @@ -45,16 +45,15 @@
 ^
danny_devito:1:2: error: stray '@' in program
 @@ -45,16 +45,15 @@
  ^
danny_devito:1:18: error: stray '@' in program
 @@ -45,16 +45,15 @@
                  ^
danny_devito:1:19: error: stray '@' in program
 @@ -45,16 +45,15 @@
                   ^
danny_devito:21:1: error: stray '@' in program
 @@ -360,8 +359,9 @@ void step(int steps)
 ^
danny_devito:21:2: error: stray '@' in program
 @@ -360,8 +359,9 @@ void step(int steps)
  ^
danny_devito:21:18: error: stray '@' in program
 @@ -360,8 +359,9 @@ void step(int steps)
                  ^
danny_devito:21:19: error: stray '@' in program
 @@ -360,8 +359,9 @@ void step(int steps)
                   ^
danny_devito:1:4: error: expected unqualified-id before '-' token
 @@ -45,16 +45,15 @@
    ^
danny_devito:22:19: error: 'i' does not name a type
   for (int i = 0; i < steps * MICROSTEPS; i++)
                   ^
danny_devito:22:43: error: 'i' does not name a type
   for (int i = 0; i < steps * MICROSTEPS; i++)
                                           ^
exit status 1
stray '@' in program

**Comment by mlemasters2002** [comment]
This is happening during compile? Have you tried the newest code yet?

**Comment by Omegaleg** [comment]
Yes this is what is happening during compile.

**Comment by Omegaleg** [comment]
Also where is the new basic code?

**Comment by mlemasters2002** [comment]
Aidan uploaded yesterday or the day before on his branch.  I have not seen that compile error.

**Comment by Omegaleg** [comment]
So where can I find it?

**Comment by mlemasters2002** [comment]
https://github.com/AidanMcCarthy/combobreaker

**Comment by Omegaleg** [comment]
Actually I worded it wrong. What I meant by basic code was the code for just the stepper motor. I wanted to make the simple version of the combo breaker.

**Comment by mlemasters2002** [comment]
I have not seen code for the simple version.  The code that we are currently using is only doing brute force, which is the same as the simple version.  Currently we are working through some accuracy issues, the code is for the most part now actuating as expected.

**Comment by Omegaleg** [comment]
so I could use the code you gave me without any modifications.

**Comment by mlemasters2002** [comment]
The code is now doing what is expect, but we are seeing drift in the numbers, enough that ti would not work after about the 12th combination tried.  As a proof of concept I think it shows it is possible, but need to do some adjusting.  There is one difference from what I am doing and what Aidan is doing.  I am using micro stepping on the step stick and Aidan is doing it as speed.  I don’t know if one is better than the other, I did it because I was researching the step stick and found how to activate, now that it is all done, I figure I might as well keep it.  I will say all notion of skipping has been eliminated since I have activated micro stepping on the hardware side.

**Comment by Omegaleg** [comment]
Well good luck to you guys and I hope you find the solution. The reason why i made this thread was to see if anyone could give me a detailed guide on how to make the simple combo breaker.

**Comment by mlemasters2002** [comment]
I have looked for it as well, but have not found it, I don’t think it was ever published.

**Comment by mlemasters2002** [comment]
If you have the simple code, send a link I can take a look.

**Comment by Omegaleg** [comment]
Here is the simple code https://github.com/samyk/combobreaker/commit/6d1320a91a40cb2a7e5487d147e57cca7abb74ca?file-filters%5B%5D=.ino#diff-825eaa5acc4bfe10efb9cae2b8edf170

@AidanMcCarthy updated it but Im still having the same issues.

**Comment by mlemasters2002** [comment]
This is the code from 2017, but I just downloaded and it compiled fine on my computer.  This still has the optical encoder in the code which does not seem to be part of the simple build based on Sam’s original videos.  Do you have the optical encoder?

**Comment by Omegaleg** [comment]
No how do I get rid of the optical encoder part. I just need the code for the stepper motor.

**Comment by Omegaleg** [comment]
So did anyone find the basic code just for the motor driver?

**Comment by AidanMcCarthy** [comment]
@Omegaleg The code doesn't exist. You would have to work on it yourself. 

It's not too hard but you need some coding knowledge. This is very much a work-in-progress and not a fully tested solution that's ready for someone to click and be up and running.
